adagrams/game.py

```diff
@@ -1,11 +1,84 @@
-def draw_letters():
-    pass
+import copy 
+import random
 
-def uses_available_letters(word, letter_bank):
-    pass
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
 
+def draw_letters():
+    i = 0
+    ten_letters = []
+    no_change_letters = LETTER_POOL.copy()
+    while i in range(10):
+        random_letter = random.choice(list(LETTER_POOL))
+        if no_change_letters[random_letter] > 0:
+            ten_letters.append(random_letter)
+            no_change_letters[random_letter] -= 1
+            i += 1
+    return ten_letters
+     
+def uses_available_letters(word, letter_bank):
+    letter_count = 0
+    letter_copy = letter_bank[:]
+    for letter in word:
+        if letter.upper() in letter_copy:
+            letter_copy.remove(letter.upper())
+            letter_count += 1
+    return letter_count == len(word)
+            
 def score_word(word):
-    pass
+    score = 0
+    word = word.upper()
+    score_dict = {"A":1, "E":1, "I":1, "O":1, "U":1, "L":1, "N":1, "R":1, "S":1, "T":1,\
```

Comment:
You could also make score_dict a constant by calling it SCORE_DICT and declare it outside the function like you did with LETTER_POOL to declutter the function. And in the future, you might have other methods that need score_dict so if it's declared outside of this function then it'll be accessible elsewhere later.